### PR TITLE
drivers: i3c: add i3c-rtio

### PR DESCRIPTION
**Pull request by XenuIsWatching**
This adds support for i3c RTIO along with adding the default calls for i3c (and i2c) for all the i3c drivers

This is heavily based of the i2c default handler. https://github.com/zephyrproject-rtos/zephyr/pull/79890

### CODE REVIEW

File: include/zephyr/rtio/rtio.h [hunk 1]
```diff
@@ -286,6 +340,17 @@ struct rtio_sqe {
...
+		/** OP_I3C_CONFIGURE */
+		struct {
+			/* enum i3c_config_type type; */
+			int type;
+			void *config;
+		} i3c_config;
+
+		/** OP_I3C_CCC */
+		/* struct i3c_ccc_payload *ccc_payload; */
+		void *ccc_payload;
```
Comment (teburd) [inline include/zephyr/rtio/rtio.h:353]: Is the payload mutable? Should potentially be const?

**Review by teburd — DISMISSED** [review]
One remark on the CCC functionality (unfamiliar with it!) that maybe could be const correct or not depending on the answer, approved regardless.

File: include/zephyr/rtio/rtio.h (hunk 1 above)
Comment (XenuIsWatching) [reply include/zephyr/rtio/rtio.h:353]: yes... there is a num_xfer with in there to return the 'true' number of bytes actually sent/received as the target can control when the End of Data occurs.... but I should probably remove this as I was trying to integrate the `do_ccc` in to the rtio... but gave up for now

Comment (teburd) [reply include/zephyr/rtio/rtio.h:353]: This sounds like some more fun things to look at in the future for sure.

Comment (XenuIsWatching) [reply include/zephyr/rtio/rtio.h:353]: yeah... thinking more about it I think i'll just keep it in for now.

File: drivers/i3c/i3c_rtio.c [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+	/* On error bail */
+	if (status < 0) {
+		rtio_iodev_sqe_err(ctx->txn_head, status);
+		return i3c_rtio_next(ctx, true);
+	}
+
+	/* Try for next submission in the transaction */
+	ctx->txn_curr = rtio_txn_next(ctx->txn_curr);
+	if (ctx->txn_curr) {
+		return true;
```
Comment (avisconti) [inline drivers/i3c/i3c_rtio.c:114]: So in this case the sqe does not get completed neither with 'ok' nor with 'err'. Is this correct?

Comment (XenuIsWatching) [reply drivers/i3c/i3c_rtio.c:114]: I would assume so?? `i2c_rtio.c` is doing the exact same thing

Comment (avisconti) [reply drivers/i3c/i3c_rtio.c:114]: Yes yes, but still I don't get it. But the code seems well written, so there should be a reason I'm not 100% getting

Comment (XenuIsWatching) [reply drivers/i3c/i3c_rtio.c:114]: I'm not really sure either... I'm being rather a coding monkey :monkey_face: here and copied most of the `i2c_rtio.c`, but only understanding just enough to make it work with i3c. Perhaps @teburd would now as the original author?

**Review by avisconti — DISMISSED** [review]
I don't see big mistakes and approve the PR. The only doubt has been left not clarified yet. Maybe @teburd can give his own opinion.

File: drivers/i3c/i3c_rtio.c (hunk 2 above)
Comment (teburd) [reply drivers/i3c/i3c_rtio.c:114]: Transactions should result in one completion, either the transaction succeeds or fails. Akin to calling i2c_transfer you only know th result of the full operation not the individual sends/recvs

**Review by teburd — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**